clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,323 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/execservice"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/checker"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/runtimeobjects"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/workerqueue"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	serviceCreationLatencyName           = "ServiceCreationLatency"
+	serviceCreationLatencyWorkers        = 10
+	defaultServiceCreationLatencyTimeout = 1 * time.Minute
+	defaultWaitTimeout                   = 10 * time.Minute
+	defaultCheckInterval                 = 10 * time.Second
+	pingBackoff                          = 1 * time.Second
+	pingChecks                           = 10
+
+	creatingPhase     = "creating"
+	ipAssigningPhase  = "ipAssigning"
+	reachabilityPhase = "reachability"
+)
+
+func init() {
+	measurement.Register(serviceCreationLatencyName, createServiceCreationLatencyMeasurement)
+}
+
+func createServiceCreationLatencyMeasurement() measurement.Measurement {
+	return &serviceCreationLatencyMeasurement{
+		selector:      measurementutil.NewObjectSelector(),
+		queue:         workerqueue.NewWorkerQueue(serviceCreationLatencyWorkers),
+		creationTimes: measurementutil.NewObjectTransitionTimes(serviceCreationLatencyName),
+		pingCheckers:  checker.NewCheckerMap(),
+	}
+}
+
+type serviceCreationLatencyMeasurement struct {
+	selector      *measurementutil.ObjectSelector
+	waitTimeout   time.Duration
+	stopCh        chan struct{}
+	isRunning     bool
+	queue         workerqueue.Interface
+	client        clientset.Interface
+	creationTimes *measurementutil.ObjectTransitionTimes
+	pingCheckers  checker.CheckerMap
+}
+
+// Execute executes service startup latency measurement actions.
+// Services can be specified by field and/or label selectors.
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+// "start" action starts observation of the services.
+// "waitForReady" waits until all services are reachable.
+// "gather" returns service created latency summary.
+// This measurement only works for services with LoadBalancer type.
+// Service of this type has to have port specified.
```

Comment:
Please remove - this is part of validation that we do in k8s apiserver, so it's always the case.

Reply:
Done.

Reply:
not yet

Reply:
Now it is, I didn't push newer version for some reason.